### PR TITLE
🐛 Add type info for `RRule`, remove conformance config exceptions

### PR DESCRIPTION
**Pull request by rsimha**
Addresses https://github.com/ampproject/amphtml/pull/22946#issuecomment-505167101

### CODE REVIEW

File: extensions/amp-date-picker/0.1/dates-list.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ export class DatesList {
       }
     }
     const rruleDates = this.rrulestrs_
-      .map(rrule => rrule.after(date))
+      .map((/** {RRule} */ rrule) => rrule.after(date))
```
Comment (jridgewell) [inline extensions/amp-date-picker/0.1/dates-list.js]: I think these would be eliminated if we properly typed `tryParseRrulestr` below.

Comment (rsimha) [reply extensions/amp-date-picker/0.1/dates-list.js]: Don't think it's possible until https://github.com/google/closure-compiler/issues/3041 is fixed.

```js
extensions/amp-date-picker/0.1/dates-list.js:83: ERROR - Possible violation: ChildNode.p.after() is unusual. Please use Node.p.insertBefore()
The type information available for this expression is too loose to ensure conformance.
      .map(rrule => rrule.after(date))
                    ^^^^^^^^^^^

extensions/amp-date-picker/0.1/dates-list.js:113: ERROR - Possible violation: ChildNode.p.before() is unusual. Please use Node.p.insertBefore()
The type information available for this expression is too loose to ensure conformance.
      const rruleDay = this.moment_(rrule.before(nextDate));
                                    ^^^^^^^^^^^^

extensions/amp-date-picker/0.1/dates-list.js:141: ERROR - Bad type annotation. Unknown type RRule
 * @return {?RRule}
             ^
```

Comment (jridgewell) [reply extensions/amp-date-picker/0.1/dates-list.js]: Just import `RRule`? We're using the code anyways, having a pure type import isn't saving us any bytes.

Comment (rsimha) [reply extensions/amp-date-picker/0.1/dates-list.js]: Importing `RRule` doesn't work trivially with the code as it's written (see errors above). I'll let you tackle this in a separate PR if you find an elegant way around it :)

**Pull request merged**